File: 프로그래머스/2/49993. 스킬트리/스킬트리.java
import java.util.*;


class Solution {
    public int solution(String skill, String[] skill_trees) {
        int answer = 0;
        
        for (final String skill_tree : skill_trees) {
            boolean canMake = true;
            
            Queue<String> queue = new LinkedList<>();
            
            for (final String s : skill.split("")) {
                queue.add(s);
            }
            
            for (String s : skill_tree.split("")) {
                if (skill.contains(s) && !s.equals(queue.remove())) {
                    canMake = false;
                    break;
                }
            }
            
            if (canMake) {
                answer++;
            }
        }
        return answer;
    }
}

File: 프로그래머스/2/49993. 스킬트리/스킬트리.py
def solution(skill, skill_trees):
    answer = 0
    
    for skill_tree in skill_trees:
        que = list(skill)

        for s in skill_tree:
            if s in skill and s != que.pop(0):
                break
        else:
            answer += 1
        
    return answer